gordo_components/builder/mlflow_utils.py

```diff
@@ -398,48 +402,40 @@ def get_spauth_kwargs() -> dict:
     )
 
 
-class DatetimeEncoder(json.JSONEncoder):
+class MetadataEncoder(json.JSONEncoder):
     """
     Encode datetime.datetime objects as strings
 
     Example
     -------
-    >>> s = json.dumps({"now":datetime.now(tz=UTC)}, cls=DatetimeEncoder, indent=4)
+    >>> s = json.dumps({"now":datetime.now(tz=UTC)}, cls=MetadataEncoder, indent=4)
     >>> s = '{"now": "2019-11-22 08:34:41.636356+"}'
     """
 
+    @staticmethod
+    def is_int_or_float(obj):
+        return any([type_str in str(type(obj)) for type_str in ("float", "int")])
+
+    @staticmethod
+    def reduce_bit_type(obj):
+        for type_cls, type_str in [(float, "float"), (int, "int")]:
+            if type_str in str(type(obj)):
+                return type_cls(obj)
+
     def default(self, obj):
+        np_float_types = (np.float, np.float16, np.float32, np.float64, np.float128)
+        np_int_types = (np.int, np.int0, np.int8, np.int16, np.int32, np.int64)
         if isinstance(obj, datetime):
             return obj.strftime("%Y-%m-%d %H:%M:%S.%f+%z")
+        # Convert numpy floats and ints to built-in python types
+        elif any([isinstance(obj, t) for t in np_float_types]):
+            return float(obj)
+        elif any([isinstance(obj, t) for t in np_int_types]):
+            return int(obj)
```

Comment:
Why not use `np.integer` and `np.floating`? Don't they cover all subtypes?

Reply:
Ah.. maybe that's why. I was trying `np.float` and `np.int` and that didn't do it.

Reply:
Bummer, doesn't work.

Reply:
Solution using `np.issubtype` now pushed.